lua/cfc_notifications/shared/sh_context.lua

```diff
@@ -0,0 +1,193 @@
+CFCNotifications._timerNameCount = 1
+CFCNotifications.contextHelpers = {}
+
+function CFCNotifications._getTimerName()
+    local newName = "notification-timer-" .. tostring( CFCNotifications._timerNameCount )
+
+    CFCNotifications._timerNameCount = CFCNotifications._timerNameCount + 1 -- How I yearn for the ++ operator
+
+    return newName
+end
+
+local function checkTypes( tab, t )
+    for k, v in pairs( tab ) do
+        if type( v ) ~= t then
+            return false
+        end
+    end
+    return true
+end
+
+local CONTEXT = {}
+CFCNotifications.Base = CONTEXT
+
+function CONTEXT:Remove()
+    if SERVER then
+        for k, v in pairs( player.GetAll() ) do
+            self:RemovePopups( v )
+        end
+        net.Start( "CFC_NotificationExists" )
+        net.WriteString( self:GetID() )
+        net.WriteBool( false )
+        net.Broadcast()
+    else
+        self:RemovePopups()
+        CFCNotifications._reloadIgnoredPanels()
+    end
+    CFCNotifications.Notifications[self:GetID()] = nil
+end
+
+function CONTEXT:RemovePopup( id, ply )
+    if SERVER then
+        self:_callClient( ply, "RemovePopup", id )
+    else
+        if id then
+            CFCNotifications._removePopupByID( id )
+        else
+            CFCNotifications._removePopupByNotificationID( self:GetID() )
+        end
+    end
+end
+
+function CONTEXT:RemovePopups( ply )
+    self:RemovePopup( nil, ply )
+end
+
+function CONTEXT:Send( filter )
+    if SERVER then
+        local players = CFCNotifications._resolveFilter( filter )
+        local valid = checkTypes( players, "Player" )
+        if not valid then
+            return -- or maybe error?
+        end
+        CFCNotifications._sendClients( players, self )
+    else
+        -- Lets not pop up the same notification more than once
+        if not self:GetAllowMultiple() and self:IsNotificationShowing( self ) then
+            return
+        end
+
+        -- Check it's allowed to show
+        if self:GetCloseable() and not self:ShouldShowNotification() then
+            return
+        end
+        -- Call something in render.lua to make a panel for this notif
```

Comment:
```suggestion

        -- TODO: Call something in render.lua to make a panel for this notif
```

is this comment still relevant?